x/dex/keeper/core_helper.go

```diff
@@ -58,35 +58,47 @@ func (k Keeper) GetOrInitPair(goCtx context.Context, token0 string, token1 strin
 	return pair
 }
 
-func (k Keeper) GetOrInitTick(goCtx context.Context, pairId string, tickIndex int64) types.Tick {
-	ctx := sdk.UnwrapSDKContext(goCtx)
-
-	tick, tickFound := k.GetTick(ctx, pairId, tickIndex)
-	if !tickFound {
-		numFees := k.GetFeeTierCount(ctx)
-		tick = types.Tick{
-			PairId:    pairId,
-			TickIndex: tickIndex,
-			TickData: &types.TickDataType{
-				// TODO: clean up tickdata proto
-				Reserve0: make([]sdk.Int, numFees),
-				Reserve1: make([]sdk.Int, numFees),
-			},
-			LimitOrderTranche0To1: &types.LimitTrancheIndexes{0, 0},
-			LimitOrderTranche1To0: &types.LimitTrancheIndexes{0, 0},
-		}
-		for i := 0; i < int(numFees); i++ {
+func (k Keeper) InitTick(ctx sdk.Context, pairId string, tickIndex int64) (types.Tick, error) {
+	price0To1, err := CalcPrice0To1(tickIndex)
+	if err != nil {
+		return types.Tick{}, err
+	}
+
+	numFees := k.GetFeeTierCount(ctx)
+	tick := types.Tick{
+		PairId:    pairId,
+		TickIndex: tickIndex,
+		Price0To1: &price0To1,
+		TickData: &types.TickDataType{
 			// TODO: clean up tickdata proto
-			tick.TickData.Reserve0[i] = sdk.ZeroInt()
-			tick.TickData.Reserve1[i] = sdk.ZeroInt()
-		}
-		k.SetTick(ctx, pairId, tick)
+			Reserve0: make([]sdk.Int, numFees),
+			Reserve1: make([]sdk.Int, numFees),
+		},
+		LimitOrderTranche0To1: &types.LimitTrancheIndexes{0, 0},
+		LimitOrderTranche1To0: &types.LimitTrancheIndexes{0, 0},
+	}
+	for i := 0; i < int(numFees); i++ {
+		// TODO: clean up tickdata proto
+		tick.TickData.Reserve0[i] = sdk.ZeroInt()
+		tick.TickData.Reserve1[i] = sdk.ZeroInt()
+	}
+	k.SetTick(ctx, pairId, tick)
+
+	token0, token1 := PairToTokens(pairId)
+	k.GetOrInitLimitOrderTranche(ctx, pairId, tickIndex, token0, 0)
+	k.GetOrInitLimitOrderTranche(ctx, pairId, tickIndex, token1, 0)
+
+	return tick, nil
+}
 
-		token0, token1 := PairToTokens(pairId)
-		k.GetOrInitLimitOrderTranche(ctx, pairId, tickIndex, token0, 0)
-		k.GetOrInitLimitOrderTranche(ctx, pairId, tickIndex, token1, 0)
+func (k Keeper) GetOrInitTick(goCtx context.Context, pairId string, tickIndex int64) (types.Tick, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	tick, tickFound := k.GetTick(ctx, pairId, tickIndex)
+	if tickFound {
+		return tick, nil
+	} else {
+		return k.InitTick(ctx, pairId, tickIndex)
```

Comment:
NewTick() without a keeper would be more idiomatic here, then save using the keeper in this function